lib/Parse/ParseStmt.cpp

```diff
@@ -368,6 +368,25 @@ ParserStatus Parser::parseBraceItems(SmallVectorImpl<ASTNode> &Entries,
       diagnose(EndOfPreviousLoc, diag::statement_same_line_without_semi)
         .fixItInsert(EndOfPreviousLoc, ";");
       // FIXME: Add semicolon to the AST?
+      Token previousToken = Lexer::getTokenAtLocation(SourceMgr, PreviousLoc);
+      Optional<tok> stmtTokMatch = None;
+      if (previousToken.getKind() == tok::identifier) {
+        SourceLoc startOfLineLocation = L->getLocForStartOfLine(SourceMgr, previousToken.getLoc());
+        SourceLoc locationOfTokenAtStartOfLine = Lexer::getTokenAtLocation(SourceMgr, startOfLineLocation).getLoc();
+        if (previousToken.getLoc() == locationOfTokenAtStartOfLine) {
+          stmtTokMatch = closestMatchingStmtTokenForIdentifier(previousToken.getText());
+        }
+      }
+      if (stmtTokMatch) {
+        Entries.pop_back();
```

Comment:
We are removing latest entry, because it is `unresolved identifier` expression from parse expression. If we will not remove it `undefined identifier` error will be emitted by Sema.